src/PureCloud.Client/Apis/RecordingApi.cs

```diff
@@ -250,4 +251,995 @@ public async Task<BatchDownloadJobSubmissionResult> CreateRecordingBatchrequestA
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<Annotation>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc>
+    public async Task<Annotation> UpdateConversationRecordingAnnotationAsync(string conversationId, string recordingId, string annotationId, Annotation body, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(conversationId);
+        ArgumentException.ThrowIfNullOrEmpty(recordingId);
+        ArgumentException.ThrowIfNullOrEmpty(annotationId);
+        ArgumentNullException.ThrowIfNull(body);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var response = await client.PutAsJsonAsync($"api/v2/conversations/{Uri.EscapeDataString(conversationId)}/recordings/{Uri.EscapeDataString(recordingId)}/annotations/{Uri.EscapeDataString(annotationId)}", body, _options.JsonSerializerOptions, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return await response.Content.ReadFromJsonAsync<Annotation>(_options.JsonSerializerOptions, cancellationToken);
+    }
+
+    /// <inheritdoc>
+    public async Task<bool> DeleteRecordingCrossplatformMediaretentionpoliciesAsync(string ids, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(ids);
+
+        var parameters = new NameValueCollection();
+        parameters.Add("ids", UriHelper.ParameterToString(ids));
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var uri = UriHelper.GetUri("api/v2/recording/crossplatform/mediaretentionpolicies", parameters);
+
+        var response = await client.DeleteAsync(uri, cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return response.IsSuccessStatusCode;
+    }
+
+    /// <inheritdoc>
+    public async Task<bool> DeleteRecordingCrossplatformMediaretentionpolicyAsync(string policyId, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(policyId);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var response = await client.DeleteAsync($"api/v2/recording/crossplatform/mediaretentionpolicies/{Uri.EscapeDataString(policyId)}", cancellationToken);
+
+        response.EnsureSuccessStatusCode();
+
+        return response.IsSuccessStatusCode;
+    }
+
+    /// <inheritdoc>
+    public async Task<bool> DeleteRecordingJobAsync(string jobId, CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(jobId);
+
+        var client = _httpClientFactory.CreateClient(PureCloudConstants.PureCloudClientName);
+
+        var response = await client.DeleteAsync($"api/v2/recording/jobs/{Uri.EscapeDataString(jobId)}", cancellationToken);
+
+        response.EnsureSuccessStatusCode();
```

Comment:
Can't use this line as it'll throw exception if the request fail. So when returning Boolean, don't invoke EnsureSucceaaStatusCode